### ISSUE TITLE
VPC peering connection data source returning VPC ID instead of peering ID

### ISSUE THREAD

**Issue by slaughtr** [terraform-providers/terraform-provider-aws#11028]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
0.12.9

### Affected Resource(s)
VPC peering accepter

* data.aws_vpc_peering_connection
* And anything trying to reference the `id` value from that

### Terraform Configuration Files

```hcl
data "aws_caller_identity" "current" {}

data "aws_vpc_peering_connection" "pc" {
  peer_vpc_id          = aws_vpc.this[0].id # here, peer_vpc_id is the accepter account's vpc ID
  peer_owner_id = data.aws_caller_identity.current.account_id # again, this is accepter account
  owner_id = var.master_account_id 
}

# added after suggestions from og-aws Slack #terraform channel - have tried many times without and various configs
provider "aws" {
  alias               = "peer"
  region              = var.aws_region
  allowed_account_ids = [var.master_account_id]
  assume_role {
    role_arn = "arn:aws:iam::${var.master_account_id}:role/${var.master_account_role_name}"
  }
}

resource "aws_vpc_peering_connection_accepter" "this" {
  count = var.master_vpc_id == "" ? 0 : 1
  provider                  = "aws.peer"
  vpc_peering_connection_id = data.aws_vpc_peering_connection.pc.id
  auto_accept = true # use this if you want it to accept
  tags = var.tags
}
```

### Debug Output

There's thousands of lines, I can provide if needed. The relevant bit (where it's getting the data.aws_vpc_peering_connection.pc values) shows that the peering connection ID is coming through
```
2019-11-26T13:09:16.906-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:     <vpcPeeringConnectionSet>
2019-11-26T13:09:16.906-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:         <item>
2019-11-26T13:09:16.906-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:             <vpcPeeringConnectionId>pcx-XXXXXXXXX</vpcPeeringConnectionId>
2019-11-26T13:09:16.906-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:             <requesterVpcInfo>
2019-11-26T13:09:16.906-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <ownerId>1234567890</ownerId>
2019-11-26T13:09:16.906-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <vpcId>vpc-XXXXXXXXXXX</vpcId>
2019-11-26T13:09:16.906-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <cidrBlock>7.0.0.0/16</cidrBlock>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <cidrBlockSet>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                     <item>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                         <cidrBlock>7.0.0.0/16</cidrBlock>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                     </item>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 </cidrBlockSet>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <region>us-west-2</region>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:             </requesterVpcInfo>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:             <accepterVpcInfo>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <ownerId>YYYYYYYYYY</ownerId>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <vpcId>vpc-YYYYYYYYY</vpcId>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <cidrBlock>13.0.0.0/16</cidrBlock>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <cidrBlockSet>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                     <item>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                         <cidrBlock>13.0.0.0/16</cidrBlock>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                     </item>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 </cidrBlockSet>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <peeringOptions>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                     <allowEgressFromLocalClassicLinkToRemoteVpc>false</allowEgressFromLocalClassicLinkToRemoteVpc>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                     <allowEgressFromLocalVpcToRemoteClassicLink>false</allowEgressFromLocalVpcToRemoteClassicLink>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                     <allowDnsResolutionFromRemoteVpc>false</allowDnsResolutionFromRemoteVpc>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 </peeringOptions>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <region>us-west-2</region>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:             </accepterVpcInfo>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:             <status>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <code>active</code>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:                 <message>Active</message>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:             </status>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:             <tagSet/>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:         </item>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4:     </vpcPeeringConnectionSet>
2019-11-26T13:09:16.907-0800 [DEBUG] plugin.terraform-provider-aws_v2.39.0_x4: </DescribeVpcPeeringConnectionsResponse>
```

### Expected Behavior

data.aws_vpc_peering_connection.pc.id should output a pcx-xxxxxx ID
### Actual Behavior

data.aws_vpc_peering_connection.pc.id outputs a VPC ID

```
Error: Error reading VPC Peering Connection: InvalidVpcPeeringConnectionId.Malformed: Invalid id: "vpc-XXXXXXXXX" (expecting "pcx-...")
```
### Steps to Reproduce

1. `terraform apply`

### Important Factoids

I'm using terragrunt on top of terraform. Master account and dev account stacks are applied completely separately.

**Comment by slaughtr** [comment]
It has since come to my attention that the provider alias bit is completely unnecessary, since I have two totally different states applied totally separately. I was just willing to try anything people were pointing me at. 

This issue also exists in Terraform v0.12.17 (figured I'd try updating to make sure it wasn't a version problem)

**Comment by slaughtr** [comment]
This issue ended up being that TF had at some point saved the bad VPC ID in the state file. I figured this out after attempting to run the `destroy` command and it failed for the same reasons. My state was so messed up I had to hand destroy my entire VPC and everything connected to it and start over. Fortunately I didn't have many resources in the VPC yet so it wasn't too bad....

`"vpc_peering_connection_id": "vpc-XXXXXXXXX"`
